nemo/collections/asr/modules/audio_modules.py

```diff
@@ -0,0 +1,686 @@
+            logging.error('Could not import torchaudio. Some features might not work.')
+
+            raise ModuleNotFoundError(
+                "torchaudio is not installed but is necessary to instantiate a {self.__class__.__name__}"
+            )
+
+        super().__init__()
+
+        # For now, assume FFT length is divisible by two
+        if fft_length % 2 != 0:
+            raise ValueError(f'fft_length = {fft_length} must be divisible by 2')
+
+        self.stft = torchaudio.transforms.Spectrogram(
+            n_fft=fft_length, hop_length=hop_length, power=power, pad_mode='constant'
+        )
+
+        # number of subbands
+        self.F = fft_length // 2 + 1
+
+    @property
+    def num_subbands(self) -> int:
+        return self.F
+
+    @property
+    def input_types(self) -> Dict[str, NeuralType]:
+        """Returns definitions of module output ports.
+        """
+        return {
+            "input": NeuralType(('B', 'C', 'T'), AudioSignal()),
+            "input_length": NeuralType(('B',), LengthsType()),
+        }
+
+    @property
+    def output_types(self) -> Dict[str, NeuralType]:
+        """Returns definitions of module output ports.
+        """
+        return {
+            "output": NeuralType(('B', 'C', 'D', 'T'), SpectrogramType()),
+            "output_length": NeuralType(('B',), LengthsType()),
+        }
+
+    @typecheck()
+    def forward(self, input: torch.Tensor, input_length: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Convert a batch of C-channel input signals
+        into a batch of complex-valued spectrograms.
+
+        Args:
+            input: Time-domain input signal with C channels, shape (B, C, T)
+            input_length: Length of valid entries along the time dimension, shape (B,)
+
+        Returns:
+            Output spectrogram with F subbands and N time frames, shape (B, C, F, N)
+            and output length with shape (B,).
+        """
+        output_length = self.get_output_length(input_length=input_length)
+
+        B, T = input.size(0), input.size(-1)
+        input = input.view(B, -1, T)
+
+        # STFT output (B, C, F, N)
+        with torch.cuda.amp.autocast(enabled=False):
+            output = self.stft(input)
+
+        # Mask padded frames
+        length_mask: torch.Tensor = make_seq_mask_like(
+            lengths=output_length, like=output, time_dim=-1, valid_ones=False
+        )
+        output = output.masked_fill(length_mask, 0.0)
+
+        return output, output_length
+
+    def get_output_length(self, input_length: torch.Tensor) -> torch.Tensor:
+        """Get length of valid frames for the output.
+
+        Args:
+            input_length: number of valid samples, shape (B,)
+
+        Returns:
+            Number of valid frames, shape (B,)
+        """
+        output_length = input_length.div(self.stft.hop_length, rounding_mode='floor').add(1).long()
+        return output_length
+
+
+class SpectrogramToAudio(NeuralModule):
```

Comment:
Why drop the "Processor" at the end from these class names? Please add them back